mana.go

```diff
@@ -0,0 +1,89 @@
+		m.ManaDecayFactorEpochsSumExponent == other.ManaDecayFactorEpochsSumExponent
+}
+
+type RewardsParameters struct {
+	// ValidatorBlocksPerSlot is the number of validation blocks that should be issued by a selected validator per slot during its epoch duties.
+	ValidatorBlocksPerSlot uint8 `serix:"0,mapKey=validatorBlocksPerSlot"`
+	// ProfitMarginExponent is used for shift operation for calculation of profit margin.
+	ProfitMarginExponent uint8 `serix:"1,mapKey=profitMarginExponent"`
+	// BootstrappinDuration is the length in epochs of the bootstrapping phase, (approx 3 years).
+	BootstrappingDuration EpochIndex `serix:"2,mapKey=bootstrappinDuration"`
+	// RewardsManaShareCoefficient is the coefficient used for calculation of initial rewards, relative to the term theta/(1-theta) from the Whitepaper, with theta = 2/3.
+	RewardsManaShareCoefficient uint64 `serix:"3,mapKey=rewardsManaShareCoefficient"`
+	// DecayBalancingConstantExponent is the exponent used for calculation of the initial reward.
+	DecayBalancingConstantExponent uint8 `serix:"4,mapKey=decayBalancingConstantExponent"`
+	// DecayBalancingConstant needs to be an integer approc  calculated based on chosen DecayBalancingConstantExponent.
+	DecayBalancingConstant uint64 `serix:"5,mapKey=decayBalancingConstant"`
+	// PoolCoefficientExponent is the exponent used for shifting operation in the pool rewards calculations.
+	PoolCoefficientExponent uint8 `serix:"6,mapKey=poolCoefficientExponent"`
+}
+
+func (r RewardsParameters) Equals(other RewardsParameters) bool {
+	return r.ValidatorBlocksPerSlot == other.ValidatorBlocksPerSlot &&
+		r.ProfitMarginExponent == other.ProfitMarginExponent && r.BootstrappingDuration == other.BootstrappingDuration &&
+		r.RewardsManaShareCoefficient == other.RewardsManaShareCoefficient &&
+		r.DecayBalancingConstantExponent == other.DecayBalancingConstantExponent &&
+		r.DecayBalancingConstant == other.DecayBalancingConstant &&
+		r.PoolCoefficientExponent == other.PoolCoefficientExponent
+}
+
+func (r RewardsParameters) TargetReward(index EpochIndex, tokenSupply uint64, generationRate, generationRateExponent, slotsPerEpochExponent uint8, api API) (Mana, error) {
+	// final reward, after bootstrapping phase
+	result, err := safemath.SafeMul(tokenSupply, r.RewardsManaShareCoefficient)
+	if err != nil {
+		return 0, ierrors.Wrap(err, "failed to calculate target reward due to tokenSupply and RewardsManaShareCoefficient multiplication overflow")
+	}
+	result, err = safemath.SafeMul(result, uint64(generationRate))
+	if err != nil {
+		return 0, ierrors.Wrap(err, "failed to calculate target reward due to multiplication with generationRate overflow")
+	}
+	subExponent, err := safemath.SafeSub(generationRateExponent, slotsPerEpochExponent)
+	if err != nil {
+		return 0, ierrors.Wrap(err, "failed to calculate target reward due to generationRateExponent - slotsPerEpochExponent subtraction overflow")
+	}
+	finalReward := result >> subExponent
```

Comment:
all these lines result in a value of `finalReward` that is constant each time you call this function. Has anyone checked if there is a better place to move them? Can this be a parameter?